opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -114,6 +114,18 @@ def shutdown(self) -> None:
         for sp in self._span_processors:
             sp.shutdown()
 
+    def force_flush(self, timeout_millis: int = 30000) -> bool:
+        deadline_ns = time_ns() + timeout_millis * 1000000
+        for sp in self._span_processors:
```

Comment:
thoughts on this approach vs something like spawning multiple threads to allow the span processors to flush concurrently?

Force flush will typically fire the exporter downstream, which in many cases is a network (e.g. DataDog, stackdriver, or jaeger). Doing this synchronously you may find the timeout is reached just for the amount of time it takes to submit to one of the exporters. Also this approach results in a strong preference to the first span processor's success over the second one. 

Might be good to document if that's the intended use case. Right now there's no mention of preference in failure scenarios.

Reply:
It is true that threading might be helpful here. But then again
1. Spawning threads makes everything more complicated.
2. shutdown has the same problem as force_flush, it just lacks timeout support.

I think if we want to add the functionality you suggest, we should add some ConcurrentMultiSpanProcessor, or AsyncMultiSpanProcessor, etc. instead of making currently very simple MultiSpanProcessor more complicated.

Reply:
@toumorokoshi, it is not that i had a concrete use case regarding the `MultiSpanProcessor` in mind when making this PR. It is just that you have to go through the `MultiSpanProcessor` when wanting to make `force_flush` available on the `TracerProvider`. 
I'll add some documentation to the `MultiSpanProcessor` and the `TracerProvider`.

About the paralellism, i'd picture a solution of either having a thread pool executor in the `...MultiSpanProcessor` or spawning new threads on every call to `force_flush` (and `shutdown`). Since the `force_flush` on the `TracerProvider` itself can be called from multiple threads i'm rather leaning towards the approach of having a thread pool executor with a continuous but fixed number of threads.
Either way, since this will introduce additional overhead i agree with @Oberon00 to implement this in a separate `...MultiSpanProcessor` class.

Reply:
@mariojonke fair enough. I think we may have to revisit the timeout behavior, but am fine with waiting until there's a clearer example of where this type of stacked timeout has issues.

Also fine with maybe have a synchronous vs async one. Here's a tracking ticket to rename the existing one to "Synchronous": https://github.com/open-telemetry/opentelemetry-python/issues/622